rust/chains/tw_ton/src/signing_request/builder.rs

```diff
@@ -47,25 +46,40 @@ impl SigningRequestBuilder {
         })
     }
 
-    /// Currently, V4R2 wallet supported only.
-    fn wallet(input: &Proto::SigningInput) -> SigningResult<TonWallet<WalletV4>> {
+    /// Currently, V4R2 and V5R1 wallets supported.
+    fn wallet(input: &Proto::SigningInput) -> SigningResult<VersionedTonWallet> {
+        // get wallet version from messages
+        let wallet_version = match input.messages.first() {
```

Comment:
It looks like this is a breaking change, and the upstream code will also need to be adjusted accordingly. Are you sure you need me to make this change?

Reply:
Yes, that would be great. From a user perspective, there will be a small breaking change but it will improve the protocol significantly as the wallet version corresponds to the whole transaction, not only one transfer message.

Reply:
Okay, I have already moved Proto::Transfer::wallet_version to Proto::SigningInput::wallet_version. All the previous comments have been addressed. Could you help review it again?